[GTK][WPE] missing URI for injected user scripts and style sheets

pgorszkowski-igalia opened the pull request:
#### 907d986d108c64d34ca622b88ec666a5152e5d18
<pre>
[GTK][WPE] missing URI for injected user scripts and style sheets
<a href="https://bugs.webkit.org/show_bug.cgi?id=272152">https://bugs.webkit.org/show_bug.cgi?id=272152</a>

Reviewed by Adrian Perez de Castro.

User scripts/style sheets without the URI are not visible in WebInspector.

With this change the URI of injected user scripts/style sheets are generated as:
user-script:incremented_identifier
user-style:incremented_identifier

* LayoutTests/userscripts/window-onerror-for-isolated-world-1-expected.txt:
* LayoutTests/userscripts/window-onerror-for-isolated-world-2-expected.txt:
* LayoutTests/userscripts/window-onerror-for-isolated-world-3-expected.txt:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/UserScript.cpp: Copied from Source/WebKit/UIProcess/API/APIUserScript.cpp.
(WebCore::generateUniqueURL):
(WebCore::UserScript::UserScript):
* Source/WebCore/page/UserScript.h:
(WebCore::UserScript::UserScript): Deleted.
* Source/WebCore/page/UserStyleSheet.cpp: Copied from Source/WebKit/UIProcess/API/APIUserScript.cpp.
(WebCore::generateUniqueURL):
(WebCore::UserStyleSheet::UserStyleSheet):
* Source/WebCore/page/UserStyleSheet.h:
* Source/WebCore/page/WorkerNavigator.cpp:
* Source/WebKit/UIProcess/API/APIUserScript.cpp:
(API::UserScript::generateUniqueURL): Deleted.
* Source/WebKit/UIProcess/API/APIUserScript.h:
* Source/WebKit/UIProcess/API/APIUserStyleSheet.cpp:
(API::UserStyleSheet::generateUniqueURL): Deleted.
* Source/WebKit/UIProcess/API/APIUserStyleSheet.h:
* Source/WebKit/UIProcess/API/C/WKUserScriptRef.cpp:
(WKUserScriptCreateWithSource):

Canonical link: <a href="https://commits.webkit.org/277470@main">https://commits.webkit.org/277470@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e101431a4173899f49174216e6abdce357f92623

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47098 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/26273 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/49741 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/49782 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43148 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/49405 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/31521 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/23734 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/38367 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/47679 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/23765 "Found 1 new test failure: fast/writing-mode/english-bt-text-with-spelling-marker.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/40594 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19674 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/21137 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41744 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5143 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/43479 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/42148 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/51657 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22122 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/18498 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/45663 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/23400 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/44667 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10529 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/24182 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23116 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitUserContent.cpp:
```diff
@@ -96,7 +108,7 @@ static inline Vector<String> toStringVector(const char* const* strv)
     _WebKitUserStyleSheet(const gchar* source, WebKitUserContentInjectedFrames injectedFrames, WebKitUserStyleLevel level, const char* const* allowList, const char* const* blockList, API::ContentWorld& world)
         : userStyleSheet(adoptRef(new API::UserStyleSheet(UserStyleSheet {
-            String::fromUTF8(source), URL { },
+            String::fromUTF8(source), URL { String::fromUTF8("user-style:")+encodeFileContent(source) },
```
Style nit:

```suggestion
            String::fromUTF8(source), URL { String::fromUTF8("user-style:") + encodeFileContent(source) },
```

mcatanzaro reviewed (commented):
This looks good, but two questions:

 * Could it be done one level lower, in API::UserStyleSheet and API::UserScript, so the macOS and GLib APIs share the same behavior in case no URL is specified?
 * What happens when two different stylesheets or scripts have the same content? It's probably OK for them to share the same URL, but does anything break?

aperezdc reviewed (approved):
This looks reasonable to me, and @mcatanzaro's suggestion of doing this in common code makes sense in my mind.

But as this now introduces a cross-port change I would prefer that someone else takes a look and rubberstamps the patch before landing, for example @achristensen07 or @weinig from Apple.

pgorszkowski-igalia committed:
[GTK][WPE] missing URI for injected user scripts and style sheets
https://bugs.webkit.org/show_bug.cgi?id=272152

Reviewed by Adrian Perez de Castro.

User scripts/style sheets without the URI are not visible in WebInspector.

With this change the URI of injected user scripts/style sheets are generated as:
user-script:incremented_identifier
user-style:incremented_identifier

* LayoutTests/userscripts/window-onerror-for-isolated-world-1-expected.txt:
* LayoutTests/userscripts/window-onerror-for-isolated-world-2-expected.txt:
* LayoutTests/userscripts/window-onerror-for-isolated-world-3-expected.txt:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/UserScript.cpp: Copied from Source/WebKit/UIProcess/API/APIUserScript.cpp.
(WebCore::generateUniqueURL):
(WebCore::UserScript::UserScript):
* Source/WebCore/page/UserScript.h:
(WebCore::UserScript::UserScript): Deleted.
* Source/WebCore/page/UserStyleSheet.cpp: Copied from Source/WebKit/UIProcess/API/APIUserScript.cpp.
(WebCore::generateUniqueURL):
(WebCore::UserStyleSheet::UserStyleSheet):
* Source/WebCore/page/UserStyleSheet.h:
* Source/WebCore/page/WorkerNavigator.cpp:
* Source/WebKit/UIProcess/API/APIUserScript.cpp:
(API::UserScript::generateUniqueURL): Deleted.
* Source/WebKit/UIProcess/API/APIUserScript.h:
* Source/WebKit/UIProcess/API/APIUserStyleSheet.cpp:
(API::UserStyleSheet::generateUniqueURL): Deleted.
* Source/WebKit/UIProcess/API/APIUserStyleSheet.h:
* Source/WebKit/UIProcess/API/C/WKUserScriptRef.cpp:
(WKUserScriptCreateWithSource):

Canonical link: https://commits.webkit.org/277470@main
